pandas/_libs/tslib.pyx

```diff
@@ -671,9 +669,7 @@ cpdef array_to_datetime(ndarray[object] values, str errors='raise',
                     # dateutil parser will return incorrect result because
                     # it will ignore nanoseconds
                     if is_raise:
-                        raise ValueError("time data {val} doesn't "
-                                         "match format specified"
-                                         .format(val=val))
+                        raise
```

Comment:
I agree this is legit. can you add a comment here about this, e.g. its OOB (even though its just a few lines above)